torchao/quantization/quant_primitives.py

```diff
@@ -632,3 +633,172 @@ def _choose_qparams_affine(
     scale = torch.clamp(scale, min=eps)
 
     return scale.to(dtype=scale_dtype), zero_point.to(dtype=zero_point_dtype)
+
+
+#HQQ
+############################################################################
+# Shrinking operator (proximal operator for the lp norm)
+def shrink_lp_op(x: torch.Tensor, beta: float, lp_norm: float) -> torch.Tensor:
```

Comment:
nit: please add `_` before ops that are not top level

Reply:
https://github.com/pytorch/ao/pull/605/commits/41b2fb0745d20f44a03b5a3670dd26c60fd3dea1